build.cmd

```diff
@@ -76,6 +77,26 @@ echo.
 :: Eval the output from probe-win1.ps1
 for /f "delims=" %%a in ('powershell -NoProfile -ExecutionPolicy RemoteSigned "& ""%__SourceDir%\Native\probe-win.ps1"""') do %%a
 
+:GetDotNetCli
+
+set "__DotNetCliPath=%__RootBinDir%\tools\cli"
+if not "%__InstallCli%"=="" (
+    if exist "%__DotNetCliPath%" (rmdir /s /q "%__DotNetCliPath%")
+    if not %ErrorLevel%==0 (
+        echo "Could not remove current installation of dotnet CLI"
+        exit /b 1
+    )
+)
+if not exist "%__DotNetCliPath%" (
+    for /f "delims=" %%a in ('powershell -NoProfile -ExecutionPolicy RemoteSigned "& "%__SourceDir%\scripts\install-cli.ps1" -installdir "%__RootBinDir%\tools""') do (
+        echo "" > nul
+    )
+)
+if not exist "%__DotNetCliPath%" (
+    echo DotNet CLI could not be downloaded or installed.
```

Comment:
Do I understand this correctly that this is only needed to run tests? If so, do we want to fail the build so early?

It might make sense to move this entire green block down to when we're done building (so that it's also skippable with 'skiptestbuild' option)